tsdoc-config/src/TSDocConfigFile.ts

```diff
@@ -0,0 +1,333 @@
+
+    this._tsdocSchema = configJson.$schema;
+    if (configJson.extends) {
+      this._extendsPaths.push(...configJson.extends);
+    }
+
+    for (const jsonTagDefinition of configJson.tagDefinitions || []) {
+      let syntaxKind: TSDocTagSyntaxKind;
+      switch (jsonTagDefinition.syntaxKind) {
+        case 'inline': syntaxKind = TSDocTagSyntaxKind.InlineTag; break;
+        case 'block': syntaxKind = TSDocTagSyntaxKind.BlockTag; break;
+        case 'modifier': syntaxKind = TSDocTagSyntaxKind.ModifierTag; break;
+        default:
+          // The JSON schema should have caught this error
+          throw new Error('Unexpected tag kind');
+      }
+      this._tagDefinitions.push(new TSDocTagDefinition({
+        tagName: jsonTagDefinition.tagName,
+        syntaxKind: syntaxKind,
+        allowMultiple: jsonTagDefinition.allowMultiple
+      }));
+    }
+  }
+
+  private _loadWithExtends(configFilePath: string, referencingConfigFile: TSDocConfigFile | undefined,
+    alreadyVisitedPaths: Set<string>): void {
+
+    if (!configFilePath) {
+      this._reportError({
+        messageId: TSDocMessageId.ConfigFileNotFound,
+        messageText: 'File not found',
+        textRange: TextRange.empty
+      });
+      return;
+    }
+
+    this._filePath = path.resolve(configFilePath);
+
+    if (!fs.existsSync(this._filePath)) {
+      this._reportError({
+        messageId: TSDocMessageId.ConfigFileNotFound,
+        messageText: 'File not found',
+        textRange: TextRange.empty
+      });
+      return;
+    }
+
+    const hashKey: string = fs.realpathSync(this._filePath);
+    if (referencingConfigFile && alreadyVisitedPaths.has(hashKey)) {
+      this._reportError({
+        messageId: TSDocMessageId.ConfigFileCyclicExtends,
+        messageText: `Circular reference encountered for "extends" field of "${referencingConfigFile.filePath}"`,
+        textRange: TextRange.empty
+      });
+      return;
+    }
+    alreadyVisitedPaths.add(hashKey);
+
+    this._loadJsonFile();
+
+    const configFileFolder: string = path.dirname(this.filePath);
+
+    for (const extendsField of this.extendsPaths) {
+      const resolvedExtendsPath: string = resolve.sync(extendsField, { basedir: configFileFolder });
+
+      const baseConfigFile: TSDocConfigFile = new TSDocConfigFile();
+
+      baseConfigFile._loadWithExtends(resolvedExtendsPath, this, alreadyVisitedPaths);
+
+      if (baseConfigFile.fileNotFound) {
+        this._reportError({
+          messageId: TSDocMessageId.ConfigFileUnresolvedExtends,
+          messageText: `Unable to resolve "extends" reference to "${extendsField}"`,
+          textRange: TextRange.empty
+        });
+      }
+
+      this._extendsFiles.push(baseConfigFile);
+
+      if (baseConfigFile.hasErrors) {
+        this._hasErrors = true;
+      }
+    }
+  }
+
+  private static _findConfigPathForFolder(folderPath: string): string {
+    if (folderPath) {
+      let foundFolder: string = folderPath;
+      for (;;) {
```

Comment:
Ew.

Reply:
Infinite loops are the One True Loop Style! (1TLS)  do/while are degenerate special cases. 😊